kartothek/io_components/stats.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import closing
+from functools import partial
+from typing import Callable, Set, Tuple
+
+import numpy as np
+import pandas as pd
+import pyarrow.parquet as pq
+from simplekv import KeyValueStore
+
+from kartothek.core.factory import DatasetFactory
+from kartothek.core.index import ExplicitSecondaryIndex
+from kartothek.core.naming import (
+    METADATA_BASE_SUFFIX,
+    METADATA_FORMAT_JSON,
+    TABLE_METADATA_FILE,
+)
+from kartothek.serialization._io_buffer import BlockBuffer
+
+StoreFactory = Callable[[], KeyValueStore]
+
+_logger = logging.getLogger(__name__)
+
+
+def get_stats_df(
+    dataset_uuid: str, store_factory: StoreFactory, workers: int = 50
+) -> pd.DataFrame:
+    """
+    Get a statistics DataFrame for a given `dataset_uuid` and `store_factory`
+    This DataFrame contains statistics the following statistics for each `.parquet` file:
+    number of rows, number of row groups and the file size in bytes.
+    It also adds ktk specific statistics like the name of the partition to the DataFrame.
+
+    Parameters
+    ----------
+    dataset_uuid : str
+        uuid of the dataset
+    store_factory : StoreFactory
+        storefactory of the dataset
+    workers : int, optional
+        threads or workers used to fetch the data, by default 50
+
+    Returns
+    -------
+    pd.DataFrame
+        A stats DataFrame with the given columns:
+        ```
+        ["key", "num_rows", "num_row_group", "file_size", "partition]
+        ```
+    """
+    ds_factory = DatasetFactory(dataset_uuid, store_factory)
+    ds_keys = list(_get_dataset_keys(ds_factory))
```

Comment:
Why implement your own get_dataset_keys function? From my understanding it parses the DatasetMetadata and returns *all* keys, i.e. also `_common_metadata` and the `by-dataset-metadata.json` although you're filtering in `get_data` again for `.parquet` files. If you're only interested in the parquet files you can also use the `dispatch_metapartitions` functionality.

I would even suggest to implement a `MetaPartition.get_statistics` method. This way you could essentially run 

```
mp_generator = dispatch_metapartitions_from_factory(...)
with ThreadPoolExecutor(max_workers=workers) as executor:
     data = executor.map(MetaPartition.get_statistics, mp_generator)
```